### PR TITLE
feat: new teleport page

### LINKED ISSUE #7363: Teleport v2 improvements
let's make some small improvements and refine the current teleport page; with more and more chains, it's becoming pretty hardly usable, and I believe it could be better 
image

## Main 
- updated file in handoff, page teleport https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=4168%3A6846&mode=design&t=QbGany6dp6MfT3aA-1

# Design 
- from top to bottom, not many changes, mainly visual and adding dropdowns instead of tabs
- keeping the teleport to your address only. In future it would be somehow merged with transfer anyway

## Top part 
image
- heading subheading, How it works, redirects to hello page: https://hello.kodadot.xyz/tutorial/teleport-bridge

## Source and destination 
image
- its needed to fill out these dropdowns. They are rounded this time
- on the right from the designs, there is more on the dropdowns
image

- I think there could be polkadot -> polkadothub by default, as mentioned here: https://github.com/kodadot/nft-gallery/issues/6740

## Asset Amount 
image
- here, you will specify the asset amount 
- right now, its fixed to the one asset we support on both networks
- under is balance with max button - this button has the same hover effects as in the profile sidebar
- the input changes according to the designs - active black stroke

## Proceed to confirmation 
image
 - big button
 - we can keep the disabled states from the old version of teleport - insufficient funds, select destination chain etc. 

## Context message 
image
- same as the old version with "(you are owner) message

That is it. Feel free to ask, thanks!

### PR DESCRIPTION
**Pull request by newraina**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 \_\_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #7363 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer?target=1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce&usdamount=0&donation=true)

#### Community participation

- [x] [Are you at KodaDot Ecosystem Telegram?](https://t.me/kodadot_eco)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

CleanShot 2023-11-05-D5WCxjMR@2x

### <samp>🤖 Generated by Copilot at 44b9d94</samp>

This pull request enhances the teleport feature by adding new components, icons, input fields and localization strings. It modifies the `Teleport.vue` component and its related files, such as `NetworkDropdown.vue`, `locales/en.json` and `utils/chain.ts`. The goal is to improve the user interface and experience of teleporting assets across networks.

### <samp>🤖 Generated by Copilot at 44b9d94</samp>

> _To teleport assets with ease_
> _We added some dropdowns and keys_
> _`NetworkDropdown.vue`_
> _And `locales/en.json` too_
> _And some icons for networks like `dot` and `kusama`_

### CODE REVIEW

**Comment by exezbcz** [comment]
hello! thank you for this pr 

some things I saw right away 


image
- only display networks possible to teleport to 


dark mode needs some work 
image

File: utils/chain.ts [hunk 1]
```diff
@@ -43,20 +43,21 @@ export const getAvailablePrefix = (prefix: string): string => {
...
-export const availablePrefixWithIcon = () => {
-  const menus = {
-    ahk: '/token/kusama_asset_hub.svg',
-    ahp: '/token/polkadot_asset_hub.svg',
-    bsx: '/token/bsx.svg',
-    snek: '/token/bsx.svg',
-    ksm: '/token/ksm.svg',
-    rmrk: '/token/ksm.svg',
-  }
+export const chainIcons = {
```
[the remark this answers is not in the document]
Comment (newraina) [reply utils/chain.ts:46]: These two don't seem like similar blocks🥲 they're different mapping

CleanShot 2023-11-05-SZsdNKnL@2x

**Comment by newraina** [comment]
@exezbcz fixed. It was a draft PR yesterday, you can review it again now

**Comment by exezbcz** [comment]
nice job! works for me, just the few little things prury mentioned

I will probably create a small followup, feel free to pickup that as well 🚀

File: components/teleport/NetworkDropdown.vue [hunk 2]
```diff
@@ -0,0 +1,71 @@
+<template>
+  <div class="w-full">
```
Comment (roiLeo) [inline components/teleport/NetworkDropdown.vue:2]:
```suggestion
  <div>
```
I don't think it's needed

File: components/teleport/NetworkDropdown.vue [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+const validateOptions = computed(() => {
+  return props.options.filter((opt) => !opt.disabled?.value)
+})
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/abstracts/variables';
+
+.dropdown-trigger {
+  height: 2.5rem;
```
Comment (roiLeo) [inline components/teleport/NetworkDropdown.vue:65]: have you tried to use `size="large"` prop on button?

File: components/teleport/Teleport.vue [hunk 4]
```diff
@@ -1,86 +1,119 @@
...
-          class="short-address">
-          {{ shortAddress(toAddress) }}
-        </a>
+    <div class="is-flex is-flex-direction-column">
+      <div class="mb-3 mt-5">{{ $t('teleport.amount') }}</div>
+
+      <div class="is-relative amount-input">
+        <NeoInput
+          v-model="amount"
+          class=""
```
Comment (roiLeo) [inline components/teleport/Teleport.vue:61]:
```suggestion
```

File: components/teleport/Teleport.vue [hunk 5]
```diff
@@ -1,86 +1,119 @@
...
-          v-safe-href="explorerUrl"
-          target="_blank"
-          rel="nofollow noopener noreferrer"
-          class="short-address">
-          {{ shortAddress(toAddress) }}
-        </a>
+    <div class="is-flex is-flex-direction-column">
+      <div class="mb-3 mt-5">{{ $t('teleport.amount') }}</div>
+
+      <div class="is-relative amount-input">
```
Comment (roiLeo) [inline components/teleport/Teleport.vue:58]: or you can wrap it with `NeoField`

**Review by roiLeo — COMMENTED** [review]
Otherwise lgtm

note: divider is not full width on mobile

Screenshot 2023-11-07 at 10-12-19 KodaDot - NFT Market Explorer

File: components/teleport/NetworkDropdown.vue (hunk 3 above)
Comment (newraina) [reply components/teleport/NetworkDropdown.vue:65]: height of large button is 3.5rem, but what is needed here is 2.5rem

File: components/teleport/Teleport.vue (hunk 5 above)
Comment (newraina) [reply components/teleport/Teleport.vue]: updated

File: components/teleport/NetworkDropdown.vue (hunk 3 above)
Comment (roiLeo) [reply components/teleport/NetworkDropdown.vue:65]: continue in #7957

**Comment by newraina** [comment]
@roiLeo 

> note: divider is not full width on mobile

it's on purpose. The dividing line in the design isn't full width:

CleanShot 2023-11-07-bXKqd2Cl@2x

File: components/teleport/Teleport.vue [hunk 6]
```diff
@@ -399,64 +447,90 @@ const sendXCM = async () => {
...
+
+.dark-mode {
+  .network-arrow:before {
+    background: $background-dark;
+  }
+}
+
+.font-normal {
+  font-weight: normal;
+}
```
Comment (roiLeo) [inline components/teleport/Teleport.vue:535]: `has-text-weight-normal`

File: components/teleport/Teleport.vue [hunk 7]
```diff
@@ -1,86 +1,120 @@
...
+    </div>
+
+    <NeoButton
+      :label="
+        insufficientBalance
+          ? $t('teleport.insufficientBalance', [currency])
+          : 'Proceed To Confirmation'
+      "
+      size="large"
+      class="is-size-6 submit-button my-5 w-full"
```
Comment (roiLeo) [inline components/teleport/Teleport.vue:103]: expanded prop to make it full, missing native-type submit?

Comment (newraina) [reply components/teleport/Teleport.vue]: updated

File: components/teleport/Teleport.vue [hunk 8]
```diff
@@ -1,86 +1,120 @@
...
+            x2="38.3536"
+            y2="9.14645"
+            stroke="currentColor" />
+          <line
+            x1="38.3293"
+            y1="8.87629"
+            x2="30.3293"
+            y2="15.8763"
+            stroke="currentColor" />
+        </svg>
```
[the remark this answers is not in the document]
Comment (newraina) [reply components/teleport/Teleport.vue:43]: `⇒` looks bad 😢 
CleanShot 2023-11-07-nt3voC6N@2x

If I put it in static, for two colors - light and dark, I need two different image files: `arrow-light.svg` and `arrow-dark.svg`. The only difference between them is the 'stroke' attribute.

The benefit of directly putting it in vue component is that I can use 	`stroke="currentColor` to automatically get the theme color.

If it's not a problem that "one arrow requires two different image files" then I can change it to a static file format.

**Comment by newraina** [comment]
@prury 

> small thing on firefox: image

Made a little change, now on my Firefox:
CleanShot 2023-11-07-vdKZj4yb@2x

**Comment by exezbcz** [comment]
@roiLeo lel, yeah you can make it full width, I think its because I resized the auto layout and left it like this. 


my bad

**Comment by newraina** [comment]
> @roiLeo lel, yeah you can make it full width, I think its because I resized the auto layout and left it like this.

@exezbcz oh I thought it was a special design. fixed

File: components/teleport/Teleport.vue (hunk 8 above)
[the remark this answers is not in the document]
Comment (roiLeo) [reply components/teleport/Teleport.vue:43]: alright you'l

> `⇒` looks bad 😢 !

could be a font problem (line width, font-size, font-family..)

> If I put it in static, for two colors - light and dark, I need two different image files: `arrow-light.svg` and `arrow-dark.svg`. The only difference between them is the 'stroke' attribute.

stroke attribute doesn't work in component or static?

> The benefit of directly putting it in vue component is that I can use `stroke="currentColor` to automatically get the theme color.
> 
> If it's not a problem that "one arrow requires two different image files" then I can change it to a static file format.

I understand, better to have one single file than 2 duplicate file

[the remark this answers is not in the document]
Comment (roiLeo) [reply components/teleport/Teleport.vue:43]: nvm keep it like this for now

**Review by roiLeo — APPROVED** [review]

**Comment by yangwao** [comment]
Thanks!
pay 70 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $70 @ 4.99 USD/DOT ~ 14.028 $DOT
🧗 [1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce](https://polkadot.subscan.io/account/1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce)
🔗 [0x89bf0f748361aaaf445dd7349596fe75937af5216ea7b04bb07a17e5f4a73cd7](https://polkadot.subscan.io/extrinsic/0x89bf0f748361aaaf445dd7349596fe75937af5216ea7b04bb07a17e5f4a73cd7)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**